src/protocol.h

```diff
@@ -18,7 +18,8 @@ class Protocol : public std::enable_shared_from_this<Protocol>
 			std::cout << "ZLIB initialization error: " << (zstream.msg ? zstream.msg : "unknown") << std::endl;
 		}
 	}
-	virtual ~Protocol() = default;
+
+	virtual ~Protocol() { deflateEnd(&zstream); };
```

Comment:
How will this behave if `deflateInit2` fails to initialize? [Documentation indicates that it returns an error code](https://refspecs.linuxbase.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/zlib-deflateend-1.html) so maybe wrap it with an if to output when it fails to end, too

Reply:
I agree with including a verification to report errors in case they occur. `Z_DATA_ERROR` and `Z_STREAM_ERROR`

Reply:
Added couts for those two errors and moved the dtor to a .cpp file while at it.